docs/configuration/secure.md

```diff
@@ -0,0 +1,571 @@
+  * [Isolate Docroot](#isolate-docroot)
+  * [OS user handling security](#os-user-handling-security)
+  * [Major security improvement: use HTTPS](#major-security-improvement--use-https)
+    + [Redirect all unencrypted traffic to HTTPS](#redirect-all-unencrypted-traffic-to-https)
+    + [Proper SSL configuration](#proper-ssl-configuration)
+    + [Serve security related Headers](#serve-security-related-headers)
+    + [Enable OCSP stapling (TLS Certificate Status Request)](#enable-ocsp-stapling--tls-certificate-status-request-)
+    + [Enable HTTP Strict Transport Security](#enable-http-strict-transport-security)
+    + [Genrate the ILIAS ```ilClientId``` cookie with ```secure``` attribute](#genrate-the-ilias----ilclientid----cookie-with----secure----attribute)
+  * [Suppress server signature and PHP version information](#suppress-server-signature-and-php-version-information)
+  * [deny access or restrict to several files or locations](#deny-access-or-restrict-to-several-files-or-locations)
+    + [ILIAS setup](#ilias-setup)
+    + [Prevent blacklisted files of beeing served by the webserver](#prevent-blacklisted-files-of-beeing-served-by-the-webserver)
+    + [Prevent execution of PHP-Code in data-directory](#prevent-execution-of-php-code-in-data-directory)
+    + [Deny Access to local Git-Directory](#deny-access-to-local-git-directory)
+  * [Integrity check of ILIAS code in docroot](#integrity-check-of-ilias-code-in-docroot)
+  * [Use WebAccessChecker](#use-webaccesschecker)
+  * [Use secure passwords](#use-secure-passwords)
+  * [Report security issues](#report-security-issues)
+
+# Hardening and Security Guidance
+
+This guideline is going to show you some best practice examples how to improve the security of your ILIAS installation, of the webserver and it's associated components.
+The ILIAS e.V. requested a documentation for a "Secure ILIAS" in march 2019.
+The section "Hardening and Security Guidance" should be removed and the security related instructions have to be mantained in a own document.
+
+## Firewall
+
+Block all traffic by default and explicitly allow only specific traffic to port 443/TCP for HTTPS secured traffic.
+For "quality of life", it is recommend to also permit 80/TCP for a [redirect to HTTPS](#redirect-all-unencrypted-traffic-to-https-1).
+
+## File Access Rights
+
+If you're an experienced admin you MAY want to use more restrictive file access rights that we RECOMMEND in this document. To make it impossible for an attacker to modify PHP files if he gains control over the web server processes, those files SHOULD be owned by ```root``` wherever possible.
+
+The only files and directories that MUST be owned/writeable by the web user are:
+
+  * ilias.ini.php
+  * data/
+  * ILIAS data dir outside of the webservers docroot
+
+All the other files and directories should be owned by ```root```, but readable by the web user (e.g. 644/755).
```

Comment:
Having the files not writeable, ok, but owned by root? If you want to make sure (beside the missing write permission) that the webserver could not alter the files make a new user/group or use "nobody". But "root" isn't a good advice (accidentally set setuid flags could even help an attacker to get full access) and would lead to problems if more than one admin with limited access rights should handle the installation.

Reply:
Also it is enough to make the files readable by the web server, and not just everyone.
As the web servers process owner usually comes with its own group, say user ``www-data`` has a group ``www-data`` with itsels as the only member, a ``chown -R root:www-data``  allows to keep the permissions set to ``640/750``. (Replace `root` by `nobody` or whoever you choose as owner of the files.)

Reply:
In the updated PR, the username is changed to a placeholder (specific) user.